module/zfs/vdev.c

```diff
@@ -4463,7 +4482,10 @@ vdev_get_stats_ex(vdev_t *vd, vdev_stat_t *vs, vdev_stat_ex_t *vsx)
 		vs->vs_configured_ashift = vd->vdev_top != NULL
 		    ? vd->vdev_top->vdev_ashift : vd->vdev_ashift;
 		vs->vs_logical_ashift = vd->vdev_logical_ashift;
-		vs->vs_physical_ashift = vd->vdev_physical_ashift;
+		if (vd->vdev_physical_ashift <= ASHIFT_MAX)
+			vs->vs_physical_ashift = vd->vdev_physical_ashift;
+		else
+			vs->vs_physical_ashift = 0;
```

Comment:
This is for vdev stats, is reporting 0 more useful than the actual value?

Reply:
It makes user-space to not complain about "suboptimal" ashift that is too big to even be set.  I later found that user-space also uses the ASHIFT_MAX define, but I am not sure it is very good in case kernel and user-space get out of sync.

Reply:
But I am open to other thoughts about this.